Tasks/AzureRmWebAppDeployment/task.json

```diff
@@ -184,6 +184,40 @@
             "helpMarkDown": "Select the option to enable msdeploy flag MSDEPLOY_RENAME_FILES_FLAG in Azure App Service application settings. The option if set enables msdeploy to rename locked files that are locked during app deployment"
         },
         {
+            "name": "ScriptType",
+            "type": "pickList",
+            "label": "Script Type",
+            "defaultValue": "",
+            "options": {
+                "": "Select Script Type",
+                "Inline Script": "Inline Script",
+                "File Path": "File Path"
+            },
+            "groupName": "AdditionalDeploymentOptions",
+            "helpMarkDown": "Type of the script: File Path or Inline Script"
+        },
+        {
+            "name": "InlineScript",
+            "type": "multiLine",
+            "label": "Inline Script",
+            "defaultValue": ":: You can provide your deployment commands here. One command per line.",
+            "groupName": "AdditionalDeploymentOptions",
+            "visibleRule": "ScriptType == Inline Script",
+            "properties": {
+                "resizable": "true",
+                "rows": "10",
+                "maxLength": "500"
+            }
+        },
+        {
+            "name": "ScriptPath",
+            "type": "filePath",
+            "label": "Script Path",
+            "required": true,
```

Comment:
Since the user selected 'ScriptPath' from pickList (previous input). We make it mandatory.